packages/adapters/examples/segmentationVolumeLocal/index.ts

```diff
@@ -0,0 +1,633 @@
+
+            await readSegmentation(file);
+        }
+
+        // Input remove
+        elInput.remove();
+    });
+    document.body.appendChild(elInput);
+
+    // Input click
+    elInput.click();
+}
+
+async function readSegmentation(file) {
+    const imageId = wadouri.fileManager.add(file);
+
+    const image = await imageLoader.loadAndCacheImage(imageId);
+
+    if (!image) {
+        return;
+    }
+
+    const instance = metaData.get("instance", imageId);
+
+    if (instance.Modality !== "SEG") {
+        console.error("This is not segmentation");
+        return;
+    }
+
+    const arrayBuffer = image.data.byteArray.buffer;
+
+    loadSegmentation(arrayBuffer);
+}
+
+async function loadSegmentation(arrayBuffer) {
+    //
+    const newSegmentationId = "LOAD_SEGMENTATION_ID:" + csUtilities.uuidv4();
+
+    //
+    const generateToolState =
+        await Cornerstone3D.Segmentation.generateToolState(
+            imageIds,
+            arrayBuffer,
+            metaData
+        );
+
+    //
+    const derivedVolume = await addSegmentationsToState(newSegmentationId);
+    //
+    const derivedVolumeScalarData = derivedVolume.getScalarData();
+    //
+    derivedVolumeScalarData.set(
+        new Uint8Array(generateToolState.labelmapBufferArray[0])
+    );
+
+    // Update the dropdown
+    updateSegmentationDropdown(newSegmentationId);
+}
+
+async function exportSegmentation() {
+    //
+    const segmentationIds = getSegmentationIds();
+    //
+    if (!segmentationIds.length) {
+        return;
+    }
+
+    // Get cache volume
+    const cacheVolume = cache.getVolume(volumeId);
+    const csImages = cacheVolume.getCornerstoneImages();
+
+    // Get active segmentation representation
+    const activeSegmentationRepresentation =
+        segmentation.activeSegmentation.getActiveSegmentationRepresentation(
+            toolGroupId
+        );
+
+    const cacheSegmentationVolume = cache.getVolume(
+        activeSegmentationRepresentation.segmentationId
+    );
+    const activeSegmentationRepresentationUid =
+        activeSegmentationRepresentation.segmentationRepresentationUID;
+
+    //
+    const labelmap = Cornerstone3D.Segmentation.generateLabelMaps2DFrom3D(
+        cacheSegmentationVolume
+    );
+
+    // Generate fake metadata as an example
+    labelmap.metadata = [];
+    labelmap.segmentsOnLabelmap.forEach(segmentIndex => {
+        const color = segmentation.config.color.getColorForSegmentIndex(
+            toolGroupId,
+            activeSegmentationRepresentationUid,
+            segmentIndex
+        );
+
+        const segmentMetadata = generateMockMetadata(segmentIndex, color);
+        labelmap.metadata[segmentIndex] = segmentMetadata;
+    });
+
+    const generatedSegmentation =
+        Cornerstone3D.Segmentation.generateSegmentation(
+            csImages,
+            labelmap,
+            metaData
+        );
+
+    downloadDICOMData(generatedSegmentation.dataset, "mySEG.dcm");
+}
```

Comment:
Seems like these are repeated, maybe you want to refactor them inside demo utils?

Reply:
Repeated because of “segmentationVolumeLocal” and “segmentationVolume”? Now just "segmentationVolume" https://github.com/cornerstonejs/cornerstone3D/pull/1177#discussion_r1548148058